src/lib.rs

```diff
@@ -104,6 +104,13 @@ pub struct OtaUpdate {
     ota_handle: esp_ota_handle_t,
 }
 
+// SAFETY: The `partition` raw pointer, which prevents the auto-impl of Send in the first place,
+//   does no refer to (shared) owned data. Dereferencing it is similarly unsafe from any task.
+//   Furthermore, we consider advancing a running OTA update from different tasks to be safe.
+//   Concurrent modification of the OTA state is prohibited by the requirement of a &mut self
+//   borrow.
```

Comment:
This is not wrong. But I would say that `OtaUpdate` not implementing `Sync` is the main thing preventing concurrent writes. Nothing in `OtaUpdate::write` uses `self` in a way that requires `&mut`, so we could probably change it to `&self` and it would still be safe. `!Sync` prevents the user from spreading references to multiple tasks anyway.

Can we have the comment say that intsead?

Reply:
Ah, yes, you are right. At the moment, both, `!Sync` *and* `&mut self` prevent concurrent operations on the OTA update. I'll remove the sentence completely, as the comment should only justify the `Send` impl and not a potential `Sync` impl, which would deal with concurrent access.

I'll also focus more on the thread-specific aspect in the first sentence of the comment.